src/emd/rt_bse_types.F

```diff
@@ -0,0 +1,757 @@
+                                                                   sim_nsteps = -1
+      REAL(kind=dp)                                             :: sim_time = 0.0_dp, &
+                                                                   sim_dt = 0.1_dp, &
+                                                                   etrs_threshold = 1.0e-7_dp, &
+                                                                   exp_accuracy = 1.0e-10_dp, &
+                                                                   ft_damping = 0.0_dp, &
+                                                                   ft_start = 0.0_dp
+      ! Which element of polarizability to print out
+      INTEGER, DIMENSION(2)                                     :: pol_element = [1, 1]
+      TYPE(dft_control_type), POINTER                           :: dft_control => NULL()
+      ! DEBUG : Trying keeping the reference to previous environments inside this one
+      TYPE(qs_environment_type), POINTER                        :: qs_env => NULL()
+      TYPE(post_scf_bandstructure_type), POINTER                :: bs_env => NULL()
+      ! Stores data needed for reading/writing to the restart files
+      TYPE(section_vals_type), POINTER                          :: restart_section => NULL(), &
+                                                                   field_section => NULL(), &
+                                                                   rho_section => NULL(), &
+                                                                   ft_section => NULL(), &
+                                                                   pol_section => NULL(), &
+                                                                   moments_section => NULL()
+      LOGICAL                                                   :: restart_extracted = .FALSE.
+
+      ! Different indices signify different spins
+      TYPE(cp_cfm_type), DIMENSION(:), POINTER                  :: ham_effective => NULL()
+      TYPE(cp_cfm_type), DIMENSION(:), POINTER                  :: ham_reference => NULL()
+      TYPE(cp_cfm_type), DIMENSION(:), POINTER                  :: ham_workspace => NULL()
+      TYPE(cp_cfm_type), DIMENSION(:), POINTER                  :: sigma_SEX => NULL()
+      TYPE(cp_fm_type), DIMENSION(:), POINTER                   :: sigma_COH => NULL(), &
+                                                                   hartree_curr => NULL()
+
+      TYPE(cp_cfm_type), DIMENSION(:), POINTER                  :: rho => NULL(), &
+                                                                   rho_new => NULL(), &
+                                                                   rho_new_last => NULL(), &
+                                                                   rho_M => NULL(), &
+                                                                   rho_orig => NULL()
+      TYPE(cp_fm_type)                                          :: S_inv_fm, &
```

Comment:
You can provide a default by setting `S_inv_fm = cp_fm_type()` and similar in other spots.

Reply:
Thank you for the idea, I was not sure how to do this properly in Fortran.